pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -17,20 +17,168 @@
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 // +resource:path=storagepool
 
+// StoragePool describes a Storage Pool resource.
 // +kubebuilder:object:root=true
-// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:scope=Namespaced,shortName=sp
 type StoragePool struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec is the specifications and configurations to be consumed by the provisioner for a StoragePool resource
+	// +required
+	Spec StoragePoolSpec `json:"spec"`
+	// Status is for handling status of StoragePool resource
+	// +optional
+	Status StoragePoolStatus `json:"status,omitempty"`
 }
 
+// StoragePoolSpec is the spec for a StoragePool resource
+type StoragePoolSpec struct {
+	// StorageCohortReference points to the StorageCohort resource, the pool is to be created upon
+	// +required
+	StorageCohortReference corev1.ObjectReference `json:"storageCohortReference"`
+
+	// StorageProvisioner refers to the pool provisioner which will be responsible for creating
+	// and managing the pool
+	// +optional
+	StorageProvisioner string `json:"storageProvisioner,omitempty"`
+
+	// Configuration points to the configuration, a custom resource, map of parameters or configmap
+	// that can be used to specify the pool and its device related configuration.
+	// +required
+	Configuration interface{} `json:"configuration"`
+
+	// Capabilities to be supported by the StoragePool
+	// +optional
+	Capabilities Capabilities `json:"capabilities,omitempty"`
+
+	// RequestedCapacity needed for the pool creation
+	// +required
+	RequestedCapacity resource.Quantity `json:"requestedCapacity"`
+}
+
+// StoragePoolStatus is for handling status of StoragePool resource
+type StoragePoolStatus struct {
+	// ReferenceResource points to the pre-existing pool resource or the created pool resource after creation of pool.
+	// +optional
+	ReferenceResource corev1.ObjectReference `json:"referenceResource,omitempty"`
```

Comment:
any reason of defining `ReferenceResource` here? How will it be used with storage pool management?

Reply:
To refer a cohort-specific pool custom resource (like a shadow CR) if one exists.

Reply:
Isn't the StoragePool is that resource? I don't see any use case of this attribute, but if you think otherwise, atleast lets make it a pointer if it's marked as optional.

Reply:
CohortStoragePool (the structure declared in this PR) is a control-plane resource. The reference resource is meant to point to an equivalent (related) data-plane resource (if one exists).

Reply:
Sure, can we make it pointer to actually assert if it's there or not? Otherwise, I think above, attributes related to ObjectReference will be store in etcd after serialization.